BlockServer/config/ioc.py

```diff
@@ -37,43 +39,53 @@ def __init__(
         name: str,
         autostart: bool = True,
         restart: bool = True,
-        component: str = None,
-        macros: Dict = None,
-        pvs: Dict = None,
-        pvsets: Dict = None,
-        simlevel: str = None,
-        remotePvPrefix: str = None,
-    ):
+        component: str | None = None,
+        macros: Dict | None = None,
+        pvs: Dict | None = None,
+        pvsets: Dict | None = None,
+        simlevel: str | None = None,
+        remotePvPrefix: str | None = None,  # noqa: N803
+    ) -> None:
         """Constructor.
 
         Args:
             name: The name of the IOC
-            autostart: Whether the IOC should automatically start/restart when the configuration is
+            autostart: Whether the IOC should automatically
+                start/restart when the configuration is
             loaded/changed
-            restart: If auto start is true, then proc serv will restart the IOC if it terminates unexpectedly
+            restart: If auto start is true, then proc serv will
+                restart the IOC if it terminates unexpectedly
             component: The component the IOC belongs to
             macros: The IOC's macros
             pvs: The IOC's PVs
             pvsets: The IOC's PV sets
             simlevel: The level of simulation
-            remotePvPrefix: The remote pv prefix
+            remotePvPrefix: The remote pv prefix,
+                has to be formatted like this to be read in properly.
         """
         self.name = name
         self.autostart = autostart
         self.restart = restart
         self.component = component
-        self.remotePvPrefix = remotePvPrefix
+        self.remote_pv_prefix = remotePvPrefix
 
         if simlevel is None:
             self.simlevel = "none"
         else:
             self.simlevel = simlevel.lower()
-
-        if macros is None:
-            self.macros = {}
-        else:
-            self.macros = macros
-
+        self.macros = {}
+        if macros is not None:
+            if isinstance(macros, dict):
```

Comment:
What condition would cause `macros` to be both not-`None` and not a `dict`? According to the type-hint this shouldn't be possible?